GherkinSyncTool.Synchronizers.AzureDevOps/Client/AzureDevopsClient.cs

```diff
@@ -205,5 +195,60 @@ private WorkItemTrackingHttpClient GetWorkItemTrackingHttpClient()
 
             return workItemTrackingHttpClient;
         }
+
+        private IList<WorkItemReference> GetListOfWorkItemsWithQuery(string query)
+        {
+            var results = new List<WorkItemReference>();
+            var workItemTrackingHttpClient = GetWorkItemTrackingHttpClient();
+            var counter = 10000;
+            var moreResults = true;
+            
+            while (moreResults)
+            {
+                // wiql - Work Item Query Language
+                var wiql = new Wiql
+                {
+                    Query = $@"{query}
+                           AND [{WorkItemFields.Id}] >= {counter - 10000} 
+                           AND [{WorkItemFields.Id}] < {counter}"
+                };
+            
+                var currentResults = workItemTrackingHttpClient.QueryByWiqlAsync(wiql, _azureDevopsSettings.Project)
+                    .Result.WorkItems.ToList();
+                
+                if (!currentResults.Any())
+                {
+                    try
+                    {
+                        results.AddRange(workItemTrackingHttpClient.QueryByWiqlAsync(new Wiql
+                        {
+                            Query = $@"{query} AND [{WorkItemFields.Id}] >= {counter}"
+                        }).Result.WorkItems.ToList());
+                        
+                        moreResults = false;
+                    }
+                    catch(Exception e)
+                    {
+                        if (e.ToString().Contains("VS402337"))
```

Comment:
What if the error message changes?

Reply:
That is the error which is thrown, when size is more, than 20000. In other situations, same thing would happen, that was happening before this code. What do you mean exactly in "error message changes" ? Do you mean that, it might change from the VSTS side ? I don't think, that it's a common practice to change exception ids, for them.